sdk/storage/azdatalake/directory/client.go

```diff
@@ -230,7 +247,7 @@ func (d *Client) Delete(ctx context.Context, options *DeleteOptions) (DeleteResp
 func (d *Client) GetProperties(ctx context.Context, options *GetPropertiesOptions) (GetPropertiesResponse, error) {
 	opts := path.FormatGetPropertiesOptions(options)
 	var respFromCtx *http.Response
-	ctxWithResp := runtime.WithCaptureResponse(ctx, &respFromCtx)
+	ctxWithResp := shared.WithCaptureBlobResponse(ctx, &respFromCtx)
 	resp, err := d.blobClient().GetProperties(ctxWithResp, opts)
```

Comment:
Why are we calling GetProperties on the blob client here, instead we could call it for the generated PathClient. This way we will get the HNS properities in the resp struct.
This might not be related to your PR changes, but we should discuss on this design.